src/common/models/problem_sets.ts

```diff
@@ -116,37 +118,46 @@ export interface ParseableQuizDates {
 }
 
 export class QuizDates extends Model {
-	private _open = 0;
-	private _due = 0;
-	private _answer = 0;
+	private _open?: number;
+	private _due?: number;
+	private _answer?: number;
```

Comment:
I don't think we want to allow any of these to be undef, no? I can see for a UserQuiz that the overrides might be undef, but it's problematic to allow this... 

Plus, this change will now require an undef check anytime we want to access these dates on the frontend. What do we gain in exchange?

Reply:
> I don't think we want to allow any of these to be undef, no? I can see for a UserQuiz that the overrides might be undef, but it's problematic to allow this...

I had some rationale for this when I switched them, but can't remember that right now (I should have added a comment). I think you're right though about having default values for the non-user versions of problem sets.  Will switch back.

Reply:
I got a little time to think about this.  I had realized that a UserQuiz and Quiz had the same date structure except that the UserQuiz could have undefined values, so that's why I switched.  I have figured out a way to do this in a better way now, so QuizDates cannot have undefined values, a UserQuizDates can have undefined values (and then a MergedQuizDates will also not have undefined values). 

I just rewrote the code this morning and will get to testing now.